### PR TITLE
fix(subscriptions): speed up refreshes and prevent Android background stalls

### PR DESCRIPTION
**Pull request by D3SOX**
Large subscription lists made refreshes slow while the New feed was visible. Android refreshes also stalled after returning Home.

## Pull Request Type

- [x] Bugfix
- [ ] Feature Implementation
- [ ] Documentation
- [ ] Other

## Related issue

Reported directly during Pixel testing; no tracked issue.

## Description

Reuse copied New-feed entries for unchanged channels, shared by Subscriptions and Home. Keep history and preference filtering reactive, including time-sensitive premiere history.

Keep Chromium active while a renderer refresh owns the Android foreground service, and use the actual activity visibility for playback and UI behavior. Store Android/web subscription caches as individual IndexedDB channel records instead of rewriting the entire NeDB log for each channel. Import existing feeds and seen state atomically, then remove the old log.

## Release note category

- [ ] Not noteworthy
- [ ] Highlights
- [ ] More improvements
- [x] Fixed bugs

## Release note

Subscription refreshes are faster with large New feeds and continue when the Android app is in the background.

## Testing

1. On desktop with a large subscription list, open New and refresh Videos. Progress should advance without prolonged UI pauses. Updated titles and new/seen indicators should remain correct in New and Home, including after changing display filters or marking videos watched.
2. Run this branch on Android with a large subscription list. Refresh Videos, return to Home, and leave the app in the background for more than a minute. Progress should continue through completion and the refresh notification should disappear.
3. Reopen the app and reload it. Cached feeds and their new/seen indicators should persist. Clear the subscription cache and reload again; the old cache must not reappear.
4. Start another refresh and cancel it. Background playback should still follow the existing screen-lock playback setting.

## Additional context

Measured on a Pixel 8 Pro with 938 subscriptions: a full video refresh completed on Home in 100.9 seconds. The same five cache updates fell from 1,853 ms to 45 ms, and the largest write fell from 27 MB to 36 KB. All 21,708 cached videos survived a reload. The separate laptop measurement was 451.8 seconds with matching backend/RSS settings.

A deterministic Electron replay of 80 channel updates with New visible fell from 39.5 seconds to 6.6 seconds, with median cache action time falling from 1,962 ms to 328 ms. The fixture has 933 channels with 36 videos each. This measures cache persistence and renderer work, not a full network refresh; the laptop has not been remeasured.

Validation includes 907 JavaScript unit tests, 41 targeted desktop New/Home E2E tests, 44 Android unit tests, eight IndexedDB regression tests, a physical-device WebView lifecycle regression, and the background refresh above. Independent Standards and Spec reviews have no remaining findings after correcting a premiere-history invalidation edge case and adding a regression test. Android remains a preview feature.

### CODE REVIEW

**Comment by D3SOX** [comment]
@coderabbitai rate limit

**Comment by D3SOX** [comment]
@coderabbitai review

File: src/datastores/browserSubscriptionCache.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    const database = await new Promise((resolve, reject) => {
+      const request = indexedDB.open(name, 1)
+      request.onupgradeneeded = () => {
+        request.result.createObjectStore('channels', { keyPath: '_id' })
+        request.result.createObjectStore('metadata')
+      }
+      request.onerror = () => reject(request.error)
+      request.onsuccess = () => resolve(request.result)
+    })
+    database.onversionchange = () => database.close()
```
[the remark this answers is not in the document]
Comment (D3SOX) [reply src/datastores/browserSubscriptionCache.js]: > [!NOTE]  
> gpt-6 responding on behalf of Nico

Fixed in 29ef0c696. Both versionchange and abnormal close now invalidate only their own cached connection promise, so a delayed close from an older connection cannot discard a newer one. Added Chromium IndexedDB regressions for reopening after deletion/versionchange, abnormal closure, and delayed old close events. Both reproduced InvalidStateError before the fix; all eight cache E2E tests pass now.

File: src/renderer/App.vue [hunk 2]
```diff
@@ -4005,19 +4006,23 @@ async function enableCapacitorIntegrations() {
...
+    setAndroidAppVisible(isActive)
     if (shouldPauseAndroidPlaybackOnAppStateChange(
       isActive,
       store.getters.getContinuePlaybackWhenScreenIsLocked
     )) {
       tabMediaCoordinator.pauseAll()
     }
   })
+  const appState = await CapacitorApp.getState()
+  setAndroidAppVisible(appState.isActive)
```
[the remark this answers is not in the document]
Comment (D3SOX) [reply src/renderer/App.vue]: > [!NOTE]  
> gpt-6 responding on behalf of Nico

Fixed in 29ef0c696. The initial getState result is applied only if no appStateChange event has arrived. Added tests running the actual integration function for both event-before-snapshot and snapshot-without-event orderings, including playback pause and cleanup. The stale overwrite reproduced before the fix; all 907 unit tests pass now.
